Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/DownloadContentScriptHandler.swift

```diff
@@ -0,0 +1,110 @@
+    self.tab = tab
+  }
+  
+  static let scriptName = "DownloadContentScript"
+  static let scriptId = UUID().uuidString
+  static let messageHandlerName = "downloadContentScript"
+  static let scriptSandbox: WKContentWorld = .defaultClient
+  static let userScript: WKUserScript? = nil
+  
+  static func downloadBlob(url: URL, tab: Tab) -> Bool {
+    let safeUrl = url.absoluteString.replacingOccurrences(of: "'", with: "%27")
+    guard url.scheme == "blob" else {
+      return false
+    }
+    
+    tab.webView?.evaluateSafeJavaScript(functionName: "window.__firefox__.download",
+                                        args: [safeUrl],
+                                        contentWorld: scriptSandbox)
+    return true
+  }
+  
+  private static func fileExtensionFromMIMEType(_ mimeType: String) -> String? {
+    if let uti = UTTypeCreatePreferredIdentifierForTag(kUTTagClassMIMEType, mimeType as CFString, nil)?.takeRetainedValue(),
+       let fileExtension = UTTypeCopyPreferredTagWithClass(uti, kUTTagClassFilenameExtension)?.takeRetainedValue() {
+      return fileExtension as String
+    }
+    return nil
+  }
+  
+  func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage, replyHandler: @escaping (Any?, String?) -> Void) {
+    defer { replyHandler(nil, nil) }
+    
+    if !verifyMessage(message: message, securityToken: UserScriptManager.securityToken) {
+      assertionFailure("Missing required security token.")
+      return
+    }
+    
+    do {
+      guard let body = message.body as? [String: Any?] else {
+        return
+      }
+      
+      let info = try JSONDecoder().decode(BlobDownloadInfo.self, from: JSONSerialization.data(withJSONObject: body))
+      guard let data = Bytes.decodeBase64(info.base64String) else {
+        return
+      }
+      
+      defer {
+        browserViewController?.pendingDownloadWebView = nil
+        Self.blobUrlForDownload = nil
+      }
+      
+      guard let requestedUrl = Self.blobUrlForDownload else {
+        log.error("\(Self.scriptName): no url was requested")
+        return
+      }
+      
+      guard requestedUrl == info.url else {
+        log.error("\(Self.scriptName): URL mismatch")
+        return
+      }
+      
+      var filename = info.url.absoluteString.components(separatedBy: "/").last ?? "data"
+      if filename.isEmpty {
+        filename = "data"
+      }
+
+      if !filename.contains(".") {
+        if let fileExtension = Self.fileExtensionFromMIMEType(info.mimeType) {
+          filename += ".\(fileExtension)"
+        }
+      }
+      
+      
+//      let response = DownloadedResourceResponse(statusCode: 200, data: data)
+//      tab?.temporaryDocument?.onDocumentDownloaded(document: response, error: nil)
+//
+//      let download = Download(filename: filename, mimeType: mimeType, size: size, data: data)
+//      browserViewController?.downloadQueue.enqueue(download)
```

Comment:
This commented out code still relevant?

Reply:
This one I might remove completely. I see that `Firefox` has it, and Brave-iOS has a partial `DownloadHelper.js`, but it has no handlers, and no usage in iOS. I thought it was a mistake, so I ported the FF code to us, but it isn't used atm anywhere. Not sure if to remove it or not.